shell/list/__tests__/workload.test.ts

```diff
@@ -0,0 +1,53 @@
+import { shallowMount } from '@vue/test-utils';
+import Workload from '@shell/list/workload.vue';
+import ResourceFetch from '@shell/mixins/resource-fetch';
+import ResourceTable from '@shell/components/ResourceTable.vue';
+import { ExtendedVue, Vue } from 'vue/types/vue';
+import { DefaultProps } from 'vue/types/options';
+
+describe('component: workload', () => {
+  it('should load the schema for the workload', () => {
+    const resource = 'workload';
+    const schema = {
+      id:         resource,
+      type:       'schema',
+      attributes: {
+        kind:       'Workload',
+        namespaced: true,
+      },
+      metadata: { name: resource },
+    };
+
+    const wrapper = shallowMount(Workload as unknown as ExtendedVue<Vue, {}, {}, {}, DefaultProps>, {
+      components: { ResourceTable },
+      mixins:     [ResourceFetch],
+      mocks:      {
+        $store: {
+          getters: {
+            currentStore:                  () => jest.fn(),
+            namespaces:                    () => jest.fn(),
+            'management/byId':             () => jest.fn(),
+            'resource-fetch/refreshFlag':  () => jest.fn(),
+            'type-map/hideBulkActionsFor': () => jest.fn(),
+            'type-map/labelFor':           () => jest.fn(),
+            'type-map/optionsFor':         () => jest.fn(),
+            'type-map/headersFor':         () => jest.fn(),
+            'prefs/get':                   () => resource,
+          }
+        },
+        $fetchState: {
+          pending: false, error: true, timestamp: Date.now()
+        },
+        $route: { params: { resource } },
+      }
+    });
+
+    expect(wrapper.vm.schema).toStrictEqual(schema);
+  });
+
+  it.todo('should display the list of workloads');
```

Comment:
Do we have a plan for these unimplemented tests? Perhaps these would benefit from being linked to an issue as well so we can follow up at a later date if needed.

Reply:
Adding issue number for the existing epic.
https://github.com/rancher/dashboard/issues/7473